includes/admin/tools.php

```diff
@@ -249,6 +254,111 @@ function edd_tools_api_keys_display() {
+	?>
+
+	<div class="postbox edd-beta-support">
+		<h3><span><?php _e( 'Enable Beta Support', 'easy-digital-downloads' ); ?></span></h3>
+		<div class="inside">
+			<p><?php _e( 'Checking any of the below checkboxes will opt you in to receive pre-release update notifications. You can opt-out at any time. Pre-release updates do not install automatically, you will still have the opportunity to ignore update notifications.', 'easy-digital-downloads' ); ?></p>
+			<form method="post" action="<?php echo admin_url( 'edit.php?post_type=download&page=edd-tools&tab=betas' ); ?>">
+				<table class="form-table edd-beta-support">
+					<tbody>
+						<?php foreach( $has_beta as $slug => $product ) : ?>
+							<tr>
+								<?php $checked = ( array_key_exists( $slug, $enabled_betas ) ? ' checked="checked"' : '' ); ?>
+								<th scope="row"><?php echo esc_html( $product ); ?></th>
+								<td>
+									<input type="checkbox" name="enabled_betas[<?php echo esc_attr( $slug ); ?>]" id="enabled_betas[<?php echo esc_attr( $slug ); ?>]"<?php echo $checked; ?> />
+									<label for="enabled_betas[<?php echo esc_attr( $slug ); ?>]"><?php printf( __( 'Get updates for pre-release versions of %s', 'easy-digital-downloads' ), $product ); ?></label>
+								</td>
+							</tr>
+						<?php endforeach; ?>
+					</tbody>
+				</table>
+				<input type="hidden" name="edd_action" value="save_enabled_betas" />
+				<?php wp_nonce_field( 'edd_save_betas_nonce', 'edd_save_betas_nonce' ); ?>
+				<?php submit_button( __( 'Save', 'easy-digital-downloads' ), 'secondary', 'submit', false ); ?>
+			</form>
+		</div>
+	</div>
+
+	<?php
+	do_action( 'edd_tools_betas_after' );
+}
+add_action( 'edd_tools_tab_betas', 'edd_tools_betas_display' );
+
+
+/**
+ * Return an array of all extensions with beta support
+ *
+ * Extensions should be added as 'extension-slug' => 'Extension Name'
+ *
+ * @since       2.6.11
+ * @return      array $extensions The array of extensions
+ */
+function edd_get_beta_enabled_extensions() {
+	return apply_filters( 'edd_beta_enabled_extensions', array() );
+}
+
+
+/**
+ * Check if a given extensions has beta support enabled
+ *
+ * @since       2.6.11
+ * @param       string $slug The slug of the extension to check
+ * @return      bool True if enabled, false otherwise
+ */
+function edd_extension_has_beta_support( $slug ) {
+	$enabled_betas = edd_get_option( 'enabled_betas', array() );
+	$return        = false;
+
+	if( array_key_exists( $slug, $enabled_betas ) ) {
+		$return = true;
+	}
+
+	return $return;
+}
+
+
+/**
+ * Save enabled betas
+ *
+ * @since       2.6.11
+ * @return      void
+ */
+function edd_tools_enabled_betas_save() {
+	if( ! wp_verify_nonce( $_POST['edd_save_betas_nonce'], 'edd_save_betas_nonce' ) ) {
+		return;
+	}
+
+	if( ! current_user_can( 'manage_shop_settings' ) ) {
+		return;
+	}
+
+	if( ! empty( $_POST['enabled_betas'] ) ) {
+		edd_update_option( 'enabled_betas', $_POST['enabled_betas'] );
```

Comment:
There are a couple challenges actually. The first being that we would need to build a callback function since there isn't an existing function that allows us to sanitize that type of data easily. The best is a `filter_var` that we can use `filter_var($string, FILTER_VALIDATE_BOOLEAN);`

The 2nd challenge is there doesn't appear to be a `value` on the input:
`									<input type="checkbox" name="enabled_betas[<?php echo esc_attr( $slug ); ?>]" id="enabled_betas[<?php echo esc_attr( $slug ); ?>]"<?php echo $checked; ?> />`

Because of that we're relying on the HTML weirdness that is, a checkbox exists in the `$_POST` with or without it. We should probably add in a `value` attribute of `1`, and then write a specific callback function that uses the above `filter_var` to give us the proper data for the save.